scripts/imgtool/image.py

```diff
@@ -288,9 +296,13 @@ def ecies_hkdf(self, enckey, plainkey):
         return cipherkey, ciphermac, pubk
 
     def create(self, key, public_key_format, enckey, dependencies=None,
-               sw_type=None, custom_tlvs=None):
+               sw_type=None, custom_tlvs=None, use_random_iv=False):
         self.enckey = enckey
 
+        if use_random_iv:
+            self.hkdf_salt = os.urandom(32)
+            self.hkdf_len  += 16 * 2   # 48 for basic scheme + 16 * 2 for random IVs
```

Comment:
Remove the extra space before `+=`

Reply:
fixed

Reply:
@utzig Well basically all the changes came from NIST requirements. Our security experts have reviewed mcuboot code in scope of secure code review of CypressBootloader, put our attention to this implementation and requested changes. We came with same arguments as yours - what is an added value of 2 randomised datas instead of one randomised data, but it is a standard recommendation/requirement.

There are no simulator changes. Unfortunately did not have much time for implementation. We made changes backwards compatible, and it would only have effect when image signed with `--use-random-iv` flag is programmed.

Reply:
There also an issue opened for this same feature https://github.com/mcu-tools/mcuboot/issues/526 as @d3zd3z pointed out, when we have discussed this.